File: search/binary_search.js
// Binary Search function
function binarySearch(arr, target) {
    let left = 0, right = arr.length - 1;
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        if (arr[mid] === target) {
            return mid;  // Found
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1; // Not found
}

// Example run
if (require.main === module) {
    let sorted = [11, 12, 22, 25, 34, 64, 90];
    let target = 25;
    let index = binarySearch(sorted, target);

    if (index !== -1) {
        console.log("Element", target, "found at index", index);
    } else {
        console.log("Element", target, "not found.");
    }
}

// Export for reuse
module.exports = binarySearch;

File: search/binary_search.py
def binary_search_recursive(arr, target, low, high):
    if low > high:
        return -1
    mid = (low + high) // 2
    if arr[mid] == target:
        return mid
    elif arr[mid] > target:
        return binary_search_recursive(arr, target, low, mid - 1)
    else:
        return binary_search_recursive(arr, target, mid + 1, high)


if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5, 6, 7]
    print("Binary Search Recursive (5):", binary_search_recursive(arr, 5, 0, len(arr) - 1))
